### ISSUE TITLE
Reimport of Serialize and Deserialize

### ISSUE THREAD

**Issue by trustyrust** [poanetwork/threshold_crypto#78]
Hi there,

The current master branch is broken because you are reimporting Serialize and Deserialize in the src/serde_impl.rs [here](https://github.com/poanetwork/threshold_crypto/blob/784c815fe799b9df4af84aa5f127fb728f3318e8/src/serde_impl.rs#L11).

Please fix.

**Comment by vkomenda** [comment]
Can you please describe your problem? The current master works and is not broken. The explanation for importing `Deserialize` and `Serialize` two times is that the line
```
use serde::{Deserialize, Deserializer, Serialize, Serializer};
```
imports the traits and the line
```
use serde_derive::{Deserialize, Serialize};
```
imports the derive macros. Therefore different constructs are imported and there is no reimport.

**Comment by trustyrust** [comment]
When trying to build my crate, I get the following error:

.cargo/registry/src/github.com-1ecc6299db9ec823/threshold_crypto-0.3.0/src/serde_impl.rs:10:20
   |
9  | use serde::{Deserialize, Deserializer, Serialize, Serializer};
   |             ----------- previous import of the macro `Deserialize` here
10 | use serde_derive::{Deserialize, Serialize};
   |                    ^^^^^^^^^^^--
   |                    |
   |                    `Deserialize` reimported here
   |                    help: remove unnecessary import
   |
   = note: `Deserialize` must be defined only once in the macro namespace of this module

**Comment by vkomenda** [comment]
Can I see your crate?

**Comment by trustyrust** [comment]
I forked threshold_crypto and upgraded to serde 1.0.89 and simply included the "derive" feature.

I would suggest not to make use of serde and serde_derive in the same file

**Comment by vkomenda** [comment]
Nightly is at 1.35.0 now. I tried `cargo clippy` with the current stable 1.33.0 and it compiled OK with no errors. Also I updated the versions of `serde_derive` to 0.6.5 and `rand` to 1.0.89 and it sill compiled with no errors.

Try stable 1.33 or the latest nightly. Do you have the same errors then?

**Comment by vkomenda** [comment]
Please try #79 to see if that fixes your issue.

**Comment by trustyrust** [comment]
@vkomenda, yes that is the same patch that applied locally to circumvent this issue.

**Comment by afck** [comment]
Thanks, I merged the fix!
(But did you try `cargo update`? I imagine that would also have resolved the problems?)

**Comment by afck** [comment]
Fixed in #79. (Please reopen if I misread.)
